src/qrules/particle.py

```diff
@@ -227,14 +247,25 @@ def _repr_pretty_(self, p: PrettyPrinter, cycle: bool) -> None:
                         p.breakable()
                         p.text(f"{attribute.name}=")
                         if isinstance(value, Parity):
-                            p.text(_to_fraction(int(value), render_plus=True))
+                            p.text(
+                                _float_as_signed_fraction_str(
+                                    int(value), render_plus=True
+                                )
+                            )
                         else:
                             p.pretty(value)  # type: ignore[attr-defined]
                         p.text(",")
             p.breakable()
             p.text(")")
 
 
+def _float_as_signed_fraction_str(value: float, render_plus: bool = False) -> str:
```

Comment:
I was curious why there is a cast to `Fraction`. Maybe a doctest would be useful:
```suggestion
def _float_as_signed_fraction_str(value: float, render_plus: bool = False) -> str:
    """Convert a float to a fraction string with optional sign.

    >>> _float_as_signed_fraction_str(1.5)
    '3/2'
    >>> _float_as_signed_fraction_str(-1.5)
    '-3/2'
    """
```

Reply:
`float_as_signed_fraction_str` is now redundant and has been deleted